### ISSUE TITLE
Further increase `BigDec` precision

### ISSUE THREAD

**Issue by jonator** [osmosis-labs/osmosis-frontend#1130]
We've gotten reports that we're running into limits in BigInteger when users work with 18 decimal tokens

**Comment by AlpinYukseloglu** [comment]
Tldr on likely source of the issue: typescript's built-in `number` type only allows for up to 53 bit ints. Likely best solution is to switch all uses of `number` on relevant stableswap components (e.g. scaling factors) to use `bigint` instead

**Comment by jonator** [comment]
Though it appears we don't use `number` in any of our calculations. The scaling factors immediately get wrapped in a `BigDec` object, which under the hood is really a `bigInteger`. Is it not just that we need to increase our `BigDec` from 36->64 precision like we have on the chain? We'd need to make sure `big-integer` pkg can handle that.

**Comment by AlpinYukseloglu** [comment]
Yes, but it seems like `number` is still the bottleneck since the inputs overflow before they can even be converted to `BigDec`!

In any case though, our `BigDec` type actually has [120 bits](https://github.com/osmosis-labs/osmosis-frontend/blob/ffdbb48b737b405fa7c7eb4fdbd1fbc9fc77b387/packages/math/src/big-dec.ts#L8) of capacity, which is much much higher than the 64 bits we need, so it cannot be the bottleneck. The 36 precision you mentioned is simply the number of decimal places we want to support and it is in line with what we have [on the chain side](https://github.com/osmosis-labs/osmosis/blob/5df8cd03bdbe660c305dc2ca1669021f76490a51/osmomath/decimal.go#L23)!

**Comment by jonator** [comment]
Putting on hold until we have more user bug report info
